transformers/pipelines.py

```diff
@@ -0,0 +1,652 @@
+            with torch.no_grad():
+                predictions = self.model(**inputs)[0]
+
+        return predictions.numpy()
+
+
+class FeatureExtractionPipeline(Pipeline):
+    """
+    Feature extraction pipeline using Model head.
+    """
+    def __call__(self, *args, **kwargs):
+        return super().__call__(*args, **kwargs).tolist()
+
+
+class TextClassificationPipeline(Pipeline):
+    """
+    Text classification pipeline using ModelForTextClassification head.
+    """
+    def __init__(self, model, tokenizer: PreTrainedTokenizer, nb_classes: int = 2):
+        super().__init__(model, tokenizer)
+
+        if nb_classes < 2:
+            raise Exception('Invalid parameter nb_classes. int >= 2 is required (got: {})'.format(nb_classes))
+        self._nb_classes = nb_classes
+
+    def __call__(self, *args, **kwargs):
+        return super().__call__(*args, **kwargs).tolist()
+
+
+class NerPipeline(Pipeline):
+    """
+    Named Entity Recognition pipeline using ModelForTokenClassification head.
+    """
+    def __init__(self, model, tokenizer: PreTrainedTokenizer):
+        super().__init__(model, tokenizer)
+
+    def __call__(self, *texts, **kwargs):
+        inputs, answers = self._args_parser(*texts, **kwargs), []
+        for sentence in inputs:
+
+            # Ugly token to word idx mapping (for now)
+            token_to_word, words = [], sentence.split(' ')
+            for i, w in enumerate(words):
+                tokens = self.tokenizer.tokenize(w)
+                token_to_word += [i] * len(tokens)
+
+            # Manage correct placement of the tensors
+            with self.device_placement():
+                tokens = self.tokenizer.encode_plus(sentence, return_attention_mask=False, return_tensors='tf' if is_tf_available() else 'pt')
+
+                # Forward
+                if is_torch_available():
+                    with torch.no_grad():
+                        entities = self.model(**tokens)[0][0].cpu().numpy()
+                else:
+                    entities = self.model(tokens)[0][0].numpy()
+
+            # Normalize scores
+            answer, token_start = [], 1
+            for idx, word in groupby(token_to_word[1:-1]):
+
+                # Sum log prob over token, then normalize across labels
+                score = np.exp(entities[token_start]) / np.exp(entities[token_start]).sum(-1, keepdims=True)
+                label_idx = score.argmax()
+
+                answer += [{
+                    'word': words[idx - 1], 'score': score[label_idx].item(), 'entity': self.model.config.id2label[label_idx]
+                }]
+
+                # Update token start
+                token_start += len(list(word))
+
+            # Append
+            answers += [answer]
+        return answers
+
+
+class QuestionAnsweringPipeline(Pipeline):
+    """
+    Question Answering pipeline using ModelForQuestionAnswering head.
+    """
+
+    class QuestionAnsweringArgumentHandler(ArgumentHandler):
```

Comment:
You can define a class inside a class but it doesn't mean you should. In fact, you shouldn't :-)

It'll be just fine at module level and it'll avoid the atrocity of `QuestionAnsweringPipeline.QuestionAnsweringArgumentHandler` — a 59-character identifier — in unit tests exercizing this class.